hat/assets/js/apps/Iaso/domains/orgUnits/reviewChanges/Filter/ReviewOrgUnitChangesFilter.tsx

```diff
@@ -57,6 +83,43 @@ export const ReviewOrgUnitChangesFilter: FunctionComponent<Props> = ({
             })) || [],
         [forms],
     );
+    const initialDataSource = useMemo(
+        () =>
+            dataSources?.find(
+                source =>
+                    source.value === defaultSourceVersion.source.id.toString(),
+            )?.value || '',
+        [dataSources, defaultSourceVersion.source.id],
+    );
```

Comment:
I think this is where the deep linking is broken. You need to check the params for a source version and use it as initial source